### PR TITLE
✨ Implement ProcessStarted Notification

### PR DESCRIPTION
**Pull request by robpal94**
---

**Changes:**

1. Add a function to let the user execute a callback, when the ProcessInstance started.

**Issues:**

Related: https://github.com/process-engine/process_engine_runtime/issues/134

PR: #75

---

See the [examples](#example) for inspiration.

## How can others test the changes?

> Describe how others can test your changes (you can remove this section for typo fixes etc.)

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.).
- [x] I've rebased the `develop` branch with my branch before finishing this PR.
- [x] I've **summarized all changes** in a list above.
- [x] I've mentioned all **PRs, which relate to this one**.
- [x] I've prefixed my Pull Request title is according to [gitmoji guide](https://gitmoji.carloscuesta.me/).

## Example

<details>
<summary>
:warning: Before merging please click here to expand and follow the guide.
</summary>
<br>

Please use `:twisted_rightwards_arrows:` at the beginning of your merge commit title.

Example 1:

<pre>
<code>
:twisted_rightwards_arrows: :bug: Fix Wrong Text Decoration at ...
</code>
</pre>

To get your commit message, just copy the first part of this pull request.

Example 2:
<pre>
<code>
**Changes:**

- Fixes Wrong Text Decoration at ...
- Fixes some typos
- ...

**Issues:**

Closes #NumberOfFixedIssue

PR: #NumberOfThisPR
</code>
</pre>
</details>

## Emojis

<details>
<summary>
Expand for a list of most used Emojis.
</summary>
<br>

Please prefix your commit messages with an Emoji.

Ref: https://gitmoji.carloscuesta.me/

| Description              | Glyphe               | Emoji  |
|--------------------------|----------------------|--------|
| Bugfix                   | `:bug:`              | 🐛     |
| Fixing Security Issues   | `:lock:`             | 🔒     |
| Configuration releated   | `:wrench:`           | 🔧     |
| Cosmetic                 | `:lipstick:`         | 💄     |
| Dependencies Downgrade   | `:arrow_down:`       | ⬇️     |
| Dependencies Upgrade     | `:arrow_up:`         | ⬆️     |
| Formatting               | `:art:`              | 🎨     |
| Improving Performance    | `:zap:`              | ⚡️      |
| Initial commit           | `:tada:`             | 🎉     |
| Linter                   | `:rotating_light:`   | 🚨     |
| Miscellaneous            | `:package:`          | 📦     |
| New Feature              | `:sparkles:`         | ✨     |
| Refactoring Code         | `:recycle:`          | ♻️      |
| Releasing / Version tags | `:bookmark:`         | 🔖     |
| Removing Stuff           | `:fire:`             | 🔥     |
| Tests                    | `:white_check_mark:` | ✅     |
| Work In Progress (WIP)   | `:construction:`     | 🚧     |

</details>

### CODE REVIEW

File: src/consumer_api_service.ts [hunk 1]
```diff
@@ -105,14 +105,12 @@ export class ConsumerApiService implements IConsumerApi {
     this._eventAggregator.subscribe(Messages.EventAggregatorSettings.messagePaths.manualTaskFinished, callback);
   }
 
-  public async onProcessTerminated(identity: IIdentity, callback: Messages.CallbackTypes.OnProcessTerminatedCallback): Promise<void> {
-    await this._iamService.ensureHasClaim(identity, this._canSubscribeToEvents);
-    this._eventAggregator.subscribe(Messages.EventAggregatorSettings.messagePaths.processTerminated, callback);
+  public onProcessStarted(callback: Messages.CallbackTypes.OnProcessStartedCallback): void {
```
Comment (cmg-dev) [inline src/consumer_api_service.ts]: Identity missing.

File: src/consumer_api_service.ts [hunk 2]
```diff
@@ -105,14 +105,12 @@ export class ConsumerApiService implements IConsumerApi {
...
+    this._eventAggregator.subscribe(Messages.EventAggregatorSettings.messagePaths.processStarted, callback);
   }
 
-  public async onProcessEnded(identity: IIdentity, callback: Messages.CallbackTypes.OnProcessEndedCallback): Promise<void> {
-    await this._iamService.ensureHasClaim(identity, this._canSubscribeToEvents);
-    this._eventAggregator.subscribe(Messages.EventAggregatorSettings.messagePaths.processEnded, callback);
+  public onProcessTerminated(callback: Messages.CallbackTypes.OnProcessTerminatedCallback): void {
```
Comment (cmg-dev) [inline src/consumer_api_service.ts]: Identity missing.

File: src/consumer_api_service.ts [hunk 3]
```diff
@@ -105,14 +105,12 @@ export class ConsumerApiService implements IConsumerApi {
     this._eventAggregator.subscribe(Messages.EventAggregatorSettings.messagePaths.manualTaskFinished, callback);
   }
 
-  public async onProcessTerminated(identity: IIdentity, callback: Messages.CallbackTypes.OnProcessTerminatedCallback): Promise<void> {
-    await this._iamService.ensureHasClaim(identity, this._canSubscribeToEvents);
```
Comment (SteffenKn) [inline src/consumer_api_service.ts]: Why did you remove the claim check?

File: src/consumer_api_service.ts [hunk 4]
```diff
@@ -105,14 +105,12 @@ export class ConsumerApiService implements IConsumerApi {
...
-    this._eventAggregator.subscribe(Messages.EventAggregatorSettings.messagePaths.processTerminated, callback);
+  public onProcessStarted(callback: Messages.CallbackTypes.OnProcessStartedCallback): void {
+    this._eventAggregator.subscribe(Messages.EventAggregatorSettings.messagePaths.processStarted, callback);
   }
 
-  public async onProcessEnded(identity: IIdentity, callback: Messages.CallbackTypes.OnProcessEndedCallback): Promise<void> {
-    await this._iamService.ensureHasClaim(identity, this._canSubscribeToEvents);
```
Comment (SteffenKn) [inline src/consumer_api_service.ts]: Why did you remove the claim check?

File: src/consumer_api_service.ts [hunk 5]
```diff
@@ -105,16 +105,36 @@ export class ConsumerApiService implements IConsumerApi {
...
+  public async onProcessStarted(identity: IIdentity, callback: Messages.CallbackTypes.OnProcessStartedCallback): Promise<void> {
     await this._iamService.ensureHasClaim(identity, this._canSubscribeToEvents);
-    this._eventAggregator.subscribe(Messages.EventAggregatorSettings.messagePaths.processTerminated, callback);
+    this._eventAggregator.subscribe(Messages.EventAggregatorSettings.messagePaths.processStarted, callback);
+  }
+
+  public async onProcessWithProcessModelIdStarted(
+                                                    identity: IIdentity,
+                                                    callback: Messages.CallbackTypes.OnProcessStartedCallback,
+                                                    processModelId: string): Promise<void> {
```
Comment (ElRaptorus) [inline src/consumer_api_service.ts]: Please fix the indentation of these parameters.

File: src/consumer_api_service.ts [hunk 6]
```diff
@@ -105,16 +105,38 @@ export class ConsumerApiService implements IConsumerApi {
...
   }
 
   public async onProcessEnded(identity: IIdentity, callback: Messages.CallbackTypes.OnProcessEndedCallback): Promise<void> {
     await this._iamService.ensureHasClaim(identity, this._canSubscribeToEvents);
     this._eventAggregator.subscribe(Messages.EventAggregatorSettings.messagePaths.processEnded, callback);
   }
 
+  public async onProcessTerminated(identity: IIdentity, callback: Messages.CallbackTypes.OnProcessTerminatedCallback): Promise<void> {
+    await this._iamService.ensureHasClaim(identity, this._canSubscribeToEvents);
+    this._eventAggregator.subscribe(Messages.EventAggregatorSettings.messagePaths.processTerminated, callback);
```
Comment (Paulomart) [inline src/consumer_api_service.ts]: I would suggest a newline before this line
